zbus_names/src/property_name.rs

```diff
@@ -0,0 +1,303 @@
+use crate::{
+  utils::{impl_str_basic, impl_try_from},
+  Error, Result,
+};
+use serde::{de, Deserialize, Serialize};
+use static_assertions::assert_impl_all;
+use std::{
+  borrow::{Borrow, Cow},
+  fmt::{self, Debug, Display, Formatter},
+  ops::Deref,
+  sync::Arc,
+};
+use zvariant::{NoneValue, OwnedValue, Str, Type, Value};
+
+/// String that identifies an [property (method or signal) name][in] on the bus.
```

Comment:
* nit: why `in`? I typically abbreviate the phrase, so `pn` would suffice.
* `(method or signal)` doesn't make sense here. I guess you based this on the copy of `MemberName` and just forgot to update the docs? :wink: 
* `an` -> `a`.

Reply:
oh, yeah, just forgot a doc-update. Done in the new commit.

Reply:
> oh, yeah, just forgot a doc-update.

no problem.

> Done in the new commit.

Actually, it's a fixup to an unmerged commit so it should be squashed into the previous commit as per the contribution guideline. The main commit contains 2 logical changes though (1. addition of new API in zbus_names and 2. the use of it in the zbus_xmlgen) so it should be split into 2.

Oh and please rebase on the main branch, not merge it in so the PR only shows commits of this PR. You'll want to rebase anyway for the squashing.

Good luck and don't feel back if any of this is new to you and get stuck. I can always help with any/all of this. Just let me know if you need help.

Reply:
Should tests go in a separate commit as well? Also, was the push for tokio-vsock-test to my branch accidently? I assume it should have gone to your main branch instead.

Reply:
> Should tests go in a separate commit as well?

Would be even better IMO but I leave that to you.

>  Also, was the push for tokio-vsock-test to my branch accidently? I assume it should have gone to your main branch instead.

This happens when you merge the main branch into your (PR) branch. You may have done that indirectly by selecting the "update the branch with main" option Github offers (I wonder if there is a way to disable that) or manually, I can't tell. 

In any case, the solution is simple: `git remote update origin && git rebase origin/main YOUR_BRANCH `. Then `git reset origin/main` to redo the commits. Use `git add -p` to stage the changes for the first commit and `git commit` to create the commit for staged changes only. Then repeat for the second commit. Once done with all that, `git push -f YOUR_REMOTE YOUR_BRANCH` to finally push your commits.

Reply:
> 
> This happens when you merge the main branch into your (PR) branch. You may have done that indirectly by selecting the "update the branch with main" option Github offers (I wonder if there is a way to disable that) or manually, I can't tell.
> 
In that case, github did something it shouldn't, because I didn't do anything to my branch via github. I branched from the main branch today and pushed without fetching, merging or pulling in between. And the push was not with github-desktop or github-cli or so, but plain git-cli.

Reply:
Very strange. I just checked and the commit in question was merged into git main 2 weeks ago. So I don't know how it ended up in your branch.

Anyway, a rebase should solve the issue.

Reply:
New force-push with the same content as before, separated into 3 commits (1 for zbus-names, one for zbus-xml and one for the tests). It seems to have solved the issue with the stray commit.

Reply:
> New force-push with the same content as before, separated into 3 commits (1 for zbus-names, one for zbus-xml and one for the tests).

Awesome! No emojis? :sob: I can add them for you, if you like?

> It seems to have solved the issue with the stray commit.

I thought it would. :)

Reply:
> Add them if you want to. ¯_(ツ)_/¯

Sure but please do add them in the future contributions. :) I also see now that you didn't quite follow the commit message guidelines/conventions. I'll fix that as well. Please don't make any changes until I'm done..

Reply:
Done! Please have a look at the commits for future reference.

Reply:
Oh, the formatting is all wrong. I'll fix that too.

Reply:
fixed! I'm done. Please resolve this thread once you've read my comments here.